chain/network/src/codec.rs

```diff
@@ -35,22 +37,29 @@ impl Encoder for Codec {
 }
 
 impl Decoder for Codec {
-    type Item = Vec<u8>;
+    type Item = Result<Vec<u8>, ReasonForBan>;
     type Error = Error;
 
-    fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Vec<u8>>, Error> {
+    fn decode(&mut self, buf: &mut BytesMut) -> Result<Option<Self::Item>, Self::Error> {
         if buf.len() < 4 {
             // not enough bytes to start decoding
             return Ok(None);
         }
+
         let mut len_bytes: [u8; 4] = [0; 4];
         len_bytes.copy_from_slice(&buf[0..4]);
         let len = u32::from_le_bytes(len_bytes);
+
+        if len > self.max_length {
+            // If this point is reached, abusive peer is banned.
+            return Ok(Some(Err(ReasonForBan::Abusive)));
```

Comment:
Rationale behind this type:

The other option was sending `Err(ReasonForBan::Abusive)`, the problem with this is that this will stop the stream of message and close the connection. We can capture this error before stopping the Peer and send ban signal, but since PeerManagerActor and PeerActor are running on separate threads BanSignal and StopSignal will compete and if StopSignal lands first the Peer will stop as desired but it won't be banned, so a malicious actor could continue with its malicious behavior.

The current design doesn't stop the stream, and when the PeerActor receives this message automatically sends the BanSignal and is stopped.